internal/witness/witness.go

```diff
@@ -226,7 +226,7 @@ func (w *Witness) Update(ctx context.Context, logID string, nextRaw []byte, cPro
 	if err := proof.VerifyConsistency(logInfo.Hasher, prev.Size, next.Size, cProof, prev.Hash, next.Hash); err != nil {
 		// Complain if the checkpoints aren't consistent.
 		counterInvalidConsistency.Inc(logID)
-		return prevRaw, status.Errorf(codes.FailedPrecondition, "failed to verify consistency proof: %v", err)
+		return prevRaw, status.Errorf(codes.Unauthenticated, "failed to verify consistency proof: %v", err)
```

Comment:
I think this is the root of my concerns with this change. Generic status codes really aren't good for trying to return a nuanced explanation. What if we returned specific errors for failure modes that are relevant to some personality?

e.g. we have `var ErrCheckpointTooOld = errors.New("checkpoint too old")` in this repo. Some more specific errors like that could help communicate these nuanced cases that need handling delicately.

Reply:
Right, sentinel errors might work, as might just directly returning HTTP status codes which match the spec, but we'd still end up with an "extra" error case whch needs to be inferred by the thing calling the witness, so I think we also need to change the signature to pass in 'oldSize'.

The mismatched API is internal anyway, and I don't _think_ such changes would mean that it would become impossible to bring the other feeders along.

I can have a shot at such a PR, but wanted to have options in case we wanted to get the fix out sooner rather than later.